### PR TITLE
Remove readline , Pass II

### PR DESCRIPTION
**Pull request by Carreau**
On top of #9684 to make things easier to read.

### CODE REVIEW

**Comment by Carreau** [comment]
Can't do that, IPykernel rely on NoOpContext.

File: IPython/core/completer.py [hunk 1]
```diff
@@ -1,50 +1,13 @@
 # encoding: utf-8
 """Word completion for IPython.
 
-This module is a fork of the rlcompleter module in the Python standard
+This module stared as fork of the rlcompleter module in the Python standard
```
Comment (minrk) [inline IPython/core/completer.py]: started

File: IPython/utils/contexts.py [hunk 2]
```diff
@@ -59,6 +61,13 @@ def __exit__(self, *exc_info):
...
 
 class NoOpContext(object):
-    """Context manager that does nothing."""
+    """
+    Deprecated
+    
+    Context manager that does nothing."""
+
+    def __init__(self):
+        warnings.warn("""NoOpContext is deprecated since IPython 5.0 """)
```
Comment (minrk) [inline IPython/utils/contexts.py]: DeprecationWarning

**Comment by minrk** [comment]
Minor comments inline, but 👍

**Comment by Carreau** [comment]
Fixed the 2 comments. Thanks for the review.

**Comment by Carreau** [comment]
@takluyver a look ?

File: IPython/core/completer.py [hunk 3]
```diff
@@ -1,50 +1,13 @@
...
-- GNU readline is also used by the built-in functions input() and
-  raw_input(), and thus these also benefit/suffer from the completer
-  features.  Clearly an interactive application can benefit by
-  specifying its own completer function and using raw_input() for all
-  its input.
-
-- When the original stdin is not a tty device, GNU readline is never
-  used, and this module (and the readline module) are silently inactive.
+This is now mostly Completer implementation made to function with
+prompt_toolkit, but that should still work with readline.
```
Comment (takluyver) [inline IPython/core/completer.py]: Should it still work with readline? How would one use it? I'd probably just remove this mention of readline; it's not a use case we're trying to support, as far as I'm concerned.

File: IPython/core/interactiveshell.py [hunk 4]
```diff
@@ -1922,10 +1880,7 @@ def showindentationerror(self):
 
     def init_readline(self):
         """Moved to terminal subclass, here only to simplify the init logic."""
```
Comment (takluyver) [inline IPython/core/interactiveshell.py]: This docstring should probably say 'deprecated' now. It's no longer in the terminal subclass, now that the terminal subclass uses PTK.

**Comment by takluyver** [comment]
Other than a couple of minor notes, :+1: !

File: IPython/core/completer.py (hunk 3 above)
Comment (Carreau) [inline IPython/core/completer.py]: Ok, so there are more code we can strip out. I'll remove the sentence and strip out more code later.

**Comment by Carreau** [comment]
> Other than a couple of minor notes, 👍 !

Done fo both.

**Comment by Carreau** [comment]
Yeah !

**Pull request merged**